src/cabinUt/cabinUt.service.ts

```diff
@@ -1,76 +1,43 @@
 import { CabinUtApi } from './cabinUt.api';
-import { CabinUtDetails } from './cabinUt.interface';
+import { getVisbookId } from './cabinUt.utils';
 
 @Injectable()
 export class CabinUtService {
-  private readonly apiClient = new CabinUtApi();
+  constructor(private readonly cabinUtApi: CabinUtApi) {}
 
-  constructor(private readonly visbookService: VisbookService) {}
-
-  async getRandomCabin(): Promise<CabinUtDetails> {
-    const cabins = await this.apiClient.getCabins();
-    const cabin = cabins[Math.floor(Math.random() * cabins.length)];
-    const { id } = cabin.node;
-    const cabinDetails = await this.apiClient.getCabinDetails(id);
-    return cabinDetails;
+  async getCabins() {
+    return this.cabinUtApi.getCabins();
   }
 
-  async isCabinAvailable(
-    cabin: CabinUtDetails,
-    checkIn: string,
-    checkOut: string,
-  ): Promise<boolean> {
-    if (!dateIsValid(checkIn) || !dateIsValid(checkOut)) {
-      console.log('invalid date(s)');
-      return false;
-    }
-
-    const cabinHasReservationsUrl = isReservationsUrl(cabin.bookingUrl);
-    const cabinVisbookId = getVisbookId(cabin.bookingUrl);
-
-    if (!cabinHasReservationsUrl && cabinVisbookId === 0) {
-      return false;
-    }
-
-    const cabinIsAvailable = await this.visbookService.isCabinAvailable(
-      cabinVisbookId,
-      checkIn,
-      checkOut,
-    );
-
-    return cabinIsAvailable;
-  }
-
-  async getRandomAvailableCabin(
-    checkIn: string,
-    checkOut: string,
-  ): Promise<CabinUtDetails | undefined> {
-    if (!dateIsValid(checkIn) || !dateIsValid(checkOut)) {
-      return undefined;
-    }
-
-    const hasExceededTimeLimit = (now: number, timeLimit: number): boolean => {
-      const elapsedTime = now - startTime;
-      if (elapsedTime >= timeLimit) {
-        return true;
-      }
-      return false;
-    };
-
-    const startTime = Date.now();
-    let cabin = await this.getRandomCabin();
-
-    while ((await this.isCabinAvailable(cabin, checkIn, checkOut)) === false) {
-      cabin = await this.getRandomCabin();
-      const ONE_MIN_IN_MILLISECONDS = 1 * 60 * 1000;
-      if (hasExceededTimeLimit(Date.now(), ONE_MIN_IN_MILLISECONDS)) {
-        return undefined;
-      }
+  async getCabinDetailsAsSupabaseCabins(
+    cabins: {
+      node: {
+        id: number;
+      };
+    }[],
+  ): Promise<Omit<SupabaseCabin, 'id'>[]> {
+    const supabaseCabins = [];
+    for (const cabin of cabins) {
+      const cabinDetails = await this.cabinUtApi.getCabinDetails(cabin.node.id);
+      const bookingUrl = cabinDetails.bookingUrl;
+      const visbookId = getVisbookId(bookingUrl);
+      const description = cabinDetails.description;
+
+      supabaseCabins.push({
+        utId: cabinDetails.id,
+        visbookId: visbookId,
+        description: description,
+        bookingUrl: bookingUrl,
+        name: cabinDetails.name,
+        lastUpdatedAt: new Date().toISOString(),
```

Comment:
A common convention is to have timestamps such as `updateAt` and `createdAt`. Following this convention means you don't need to think too much about the naming. Suggest renaming this to `updatedAt`.

Reply:
Nice, thank you for the explanation. Renamed in code and database!